jenkins/ruby/distribution-build.jenkinsfile

```diff
@@ -0,0 +1,99 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    options {
+        timeout(time: 1, unit: 'HOURS')
+    }
+    agent none
+    parameters {
+        string(
+            name: 'REPOSITORY_NAME',
+            description: 'Repository to release, e.g. opensearch-ruby',
+            trim: true
+        )
+        string(
+            name: 'RELEASE_BRANCH',
+            description: 'Branch name to release, e.g. 1.2.0',
+            trim: true
+        )
+        string(
+            name: 'RUBY_INTERPRETER',
+            description: 'ruby, jruby, etc',
+            defaultValue: 'ruby-2.7.0',
+            trim: true
+        )
+    }
+    stages {
+        stage('ruby-build-sign-upload') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+                    image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211130'
+                    args '-u root' // Needed for installing RUBY_INTERPRETER 
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    git url: "https://github.com/opensearch-project/$REPOSITORY.git", branch: "$RELEASE_BRANCH"
+                    withCredentials([string(credentialsId: 'opensearchproject-rubygems-private-key', variable: 'RUBYGEMS_PRIVATE_KEY')]) {
+                        writeFile file: 'gem-private_key.pem', text: RUBYGEMS_PRIVATE_KEY
+                    }
+                    sh'''
+                        rvm install $RUBY_INTERPRETER
+                        rvm --default use $RUBY_INTERPRETER
+                        gem cert --add certs/opensearch-rubygems.pem
```

Comment:
Note: This means any ruby project would need to have the public key in exact same location (certs/opensearch-rubygems.pem)
opensearch-ruby has it [there](https://github.com/opensearch-project/opensearch-ruby/blob/main/certs/opensearch-rubygems.pem)

Reply:
Yes, lets try to start with this as the convention, we can revisit if this becomes problematic, FYI - logstash-output-opensearch was OK with this

Reply:
Sounds good. Thanks!